homeassistant/helpers/update_coordinator.py

```diff
@@ -143,11 +143,13 @@ def async_update_listeners(self) -> None:
 
     @callback
     def _unschedule_refresh(self) -> None:
-        """Unschedule any pending refresh since there is no longer any listeners."""
+        """Unschedule any pending refresh, and cancel debounce."""
         if self._unsub_refresh:
             self._unsub_refresh()
             self._unsub_refresh = None
 
+        self._debounced_refresh.async_cancel()
```

Comment:
In the four instances of `self._unsub_refresh()` in the file, two cancelled the debounce and two didn't.
I think this was a bug and all four instances should cancel the debounce.

Now all four instances call the same `_unschedule_refresh` routine.